compiler/ir/backend.jvm/src/org/jetbrains/kotlin/backend/jvm/codegen/IrTypeMapper.kt

```diff
@@ -57,9 +58,15 @@ class IrTypeMapper(val kotlinTypeMapper: KotlinTypeMapper) {
     fun mapType(irType: IrType, sw: JvmSignatureWriter, mode: TypeMappingMode) =
         kotlinTypeMapper.mapType(irType.toKotlinType(), sw, mode)
 
+    fun mapTypeAsDeclaration(irType: IrType) =
```

Comment:
It seems hard to extract the part of the code that is actually affected (name mangling) without a lot of refactoring in the TypeMapper. Since there is a rewrite in progress I would like to hold off on that.